client/pages/details.jsx

```diff
@@ -60,6 +62,10 @@ export default class Details extends React.Component {
       .then(result => {
         const test = this.state.comments.slice();
         test.push(result[0]);
+
+        const userName = jwtDecode(window.localStorage.getItem('react-context-jwt'));
+        test[test.length - 1].username = userName.username;
```

Comment:
This component is already receiving a user prop, use that instead of digging for and decoding the token.